otx/api/utils/tiler.py

```diff
@@ -270,6 +278,144 @@ def merge_results(self, results: List[Dict], shape: List[int]):
                 detections = *Tiler.detection2tuple(detections), masks
         return detections
 
+    def merge_features(
+        self, features: List, predictions: Union[Tuple, np.ndarray]
+    ) -> Union[Tuple[None, None], List[np.ndarray]]:
+        """Merge tile-level feature vectors to image-level features.
+
+        Args:
+            features: tile-level features.
+            predictions: predictions with masks for whole image.
+
+        Returns:
+            image_vector (np.ndarray): Merged feature vector for entire image.
+            image_saliency_map (List): Merged saliency map for entire image
+        """
+        if len(features) == 0:
+            return (None, None)
+        image_vector = self.merge_vectors(features)
+
+        (_, image_saliency_map), _ = features[0]
+        if isinstance(image_saliency_map, np.ndarray):
+            image_saliency_map = self.merge_maps(features)
+        else:
+            # if saliency maps weren't return from hook (Mask RCNN case)
+            image_saliency_map = self.get_tiling_saliency_map_from_segm_masks(predictions)
+
+        return image_vector, image_saliency_map
+
+    def merge_vectors(self, features: List) -> np.ndarray:
+        """Merge tile-level feature vectors to image-level feature vector.
+
+        Args:
+            features: tile-level features.
+
+        Returns:
+            merged_vectors (np.ndarray): Merged vectors for entire image.
+        """
+        vectors = [vector for (vector, _), _ in features]
+        return np.average(vectors, axis=0)
+
+    def merge_maps(self, features: List) -> np.ndarray:
+        """Merge tile-level saliency maps to image-level saliency map.
+
+        Args:
+            features: tile-level features ((vector, map), tile_meta).
+            Each saliency map is a list of maps for each detected class or None if class wasn't detected.
+
+        Returns:
+            merged_maps (np.ndarray): Merged saliency maps for entire image.
+        """
+
+        (_, image_saliency_map), image_meta = features[0]
+
+        num_classes = len(image_saliency_map)
+        feat_h, feat_w = image_saliency_map[0].shape
+        dtype = image_saliency_map[0][0].dtype
+
+        image_h, image_w, _ = image_meta["original_shape"]
+        ratio = np.array([feat_h, feat_w]) / self.tile_size
+
+        image_map_h = int(image_h * ratio[0])
+        image_map_w = int(image_w * ratio[1])
+        merged_map = [np.zeros((image_map_h, image_map_w)) for _ in range(num_classes)]
+
+        for (_, saliency_map), meta in features[1:]:
+            for class_idx in range(num_classes):
+                if saliency_map[class_idx] is None:
```

Comment:
Can this happen? Is saliency map a list?

Reply:
Actually, you're right, it can't happened, I removed this altogether with the other checks if the saliency map for certain classes is None. 
The saliency map is `np.array` here, but I made the implementation more aligned with the implementation for PyTorch (tiling.py), where the saliency map can be a `np.array `(detection case) and a `list` (instance segmentation case).
To improve the performance, I removed extra iteration through classes and process saliency map as a `np.array`